internal/resources/audit_log_settings.go

```diff
@@ -0,0 +1,323 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"net/http"
+)
+
+var (
+	_ resource.Resource                = &AuditLogSettings{}
+	_ resource.ResourceWithConfigure   = &AuditLogSettings{}
+	_ resource.ResourceWithImportState = &AuditLogSettings{}
+)
+
+// AuditLogSettings is the audit log settings resource implementation.
+type AuditLogSettings struct {
+	*providerschema.Data
+}
+
+func NewAuditLogSettings() resource.Resource {
+	return &AuditLogSettings{}
+}
+
+// Metadata returns the audit log settings resource type name.
+func (a *AuditLogSettings) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_auditlogsettings"
+}
+
+// Schema defines the schema for the audit log settings resource.
+func (a *AuditLogSettings) Schema(ctx context.Context, rsc resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = AuditLogSettingsSchema()
+}
+
+// Configure adds the provider configured client to the audit log settings resource.
+func (a *AuditLogSettings) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	a.Data = data
+}
+
+// AuditLogSettings does not have create endpoint
+// as a workaround, create has same behavior as update
+func (a *AuditLogSettings) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state providerschema.ClusterAuditSettings
+	diags := req.Plan.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
```

Comment:
We should have a function like this, as we have in the create function of other resources.

```
	err := a.validateCreateApiKeyRequest(plan)
	if err != nil {
		resp.Diagnostics.AddError(
			"Error creating ApiKey",
			"Could not create ApiKey, unexpected error:"+err.Error(),
		)
		return
	}
```